tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -151,6 +156,177 @@ public async Task EndpointsConfiguredCallback_MessagePublished()
+
+            kafkaTestingHelper.Spy.OutboundEnvelopes.Should().HaveCount(1);
+        }
+
+        [Fact]
+        public async Task EndpointsConfiguredKafkaEndOfTopicPartitionReachedCallback_WithProducingSeveralMessages()
+        {
+            var messageEventOne = new TestEventOne
+            {
+                Content = "Hello E2E!"
+            };
+
+            var messageEventTwo = new TestEventTwo
+            {
+                Content = "Hello E2E!"
+            };
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .AddSingletonBrokerCallbackHandler<KafkaEndOfTopicPartitionReachedCallback>()
+                        .WithConnectionToMessageBroker(
+                            options => options
+                                .AddMockedKafka(options => options.WithDefaultPartitionsCount(2)))
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<TestEventOne>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName, 0))
+                                .AddOutbound<TestEventTwo>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName, 1))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                                config.EnablePartitionEof = true;
+                                            })))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var kafkaTestingHelper = Host.ServiceProvider.GetRequiredService<IKafkaTestingHelper>();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            var callbackHandlers = Host.ScopedServiceProvider
+                .GetServices<IBrokerCallback>()
+                .Where(service => service is KafkaEndOfTopicPartitionReachedCallback)
+                .ToList();
+
+            callbackHandlers.Should().HaveCount(1);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 2);
```

Comment:
In these tests, or in a new one, you should also check that you receive the EOF message with the expected parameters (so one per each partition in the topic).